agents-for-bedrock/use-case-examples/product-review-agent/main.ipynb

```diff
@@ -0,0 +1,2118 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Create IAM policies for Lambda to access Bedrock Knowledge base API\n",
+    "lambda_allow_kb_policy_statement = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": [\n",
+    "\t\t\t\t\"bedrock:RetrieveAndGenerate\",\n",
+    "                \"bedrock:Retrieve\"\n",
+    "\t\t\t],\n",
+    "            \"Resource\": knowledge_base_arn\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "# Create IAM policies for Lambda to invoke Bedrock model\n",
+    "lambda_allow_invoke_policy_statement = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Sid\": \"AmazonBedrockAgentBedrockFoundationModelPolicy\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": \"bedrock:InvokeModel\",\n",
+    "            \"Resource\": [\n",
+    "                f\"arn:aws:bedrock:{region}::foundation-model/anthropic.claude-3-haiku-20240307-v1:0\"\n",
+    "            ]\n",
+    "        }\n",
+    "    ]\n",
+    "}\n",
+    "\n",
+    "lambda_kb_policy_json = json.dumps(lambda_allow_kb_policy_statement)\n",
+    "\n",
+    "lambda_kb_policy = iam_client.create_policy(\n",
+    "    PolicyName=lambda_kb_allow_policy_name,\n",
+    "    PolicyDocument=lambda_kb_policy_json\n",
+    ")\n",
+    "\n",
+    "iam_client.attach_role_policy(\n",
+    "    RoleName=lambda_role_name,\n",
+    "    PolicyArn=lambda_kb_policy['Policy']['Arn']\n",
+    ")\n",
+    "\n",
+    "lambda_invoke_policy_json = json.dumps(lambda_allow_invoke_policy_statement)\n",
+    "\n",
+    "lambda_invoke_policy = iam_client.create_policy(\n",
+    "    PolicyName=lambda_invoke_allow_policy_name,\n",
+    "    PolicyDocument=lambda_invoke_policy_json\n",
+    ")\n",
+    "\n",
+    "iam_client.attach_role_policy(\n",
+    "    RoleName=lambda_role_name,\n",
+    "    PolicyArn=lambda_invoke_policy['Policy']['Arn']\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Allow Bedrock Agent to invoke LLM"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create IAM policies for agent\n",
+    "bedrock_agent_bedrock_allow_policy_statement = {\n",
+    "    \"Version\": \"2012-10-17\",\n",
+    "    \"Statement\": [\n",
+    "        {\n",
+    "            \"Sid\": \"AmazonBedrockAgentBedrockFoundationModelPolicy\",\n",
+    "            \"Effect\": \"Allow\",\n",
+    "            \"Action\": \"bedrock:InvokeModel\",\n",
+    "            \"Resource\": [\n",
+    "                f\"arn:aws:bedrock:{region}::foundation-model/anthropic.claude-3-haiku-20240307-v1:0\"\n",
```

Comment:
can you parametrize the model here? in case the user wants to use sonnet or a another model